src/http.c

```diff
@@ -950,7 +953,9 @@ static void http_cb(struct mg_connection *c, int ev, void *evd, void *fnd) {
       bool next = false;
       int hlen = mg_http_parse((char *) c->recv.buf, c->recv.len, &hm);
       if (hlen < 0) {
-        mg_error(c, "HTTP parse:\n%.*s", (int) c->recv.len, c->recv.buf);
+        mg_printf(c, "HTTP/1.0 400 Bad Request\r\n\r\n");
+        MG_ERROR(("%lu HTTP malformed message", c->id));
+        mg_iobuf_del(&c->recv, 0, hm.message.len);
```

Comment:
I would keep the former message as close as possible.
We are no longer closing the connection as before, is this OK/better ?